### PR TITLE
WFCORE-135: remove custom runtime permissions

### PR DESCRIPTION
**Pull request by emmartins**

### CODE REVIEW

**Comment by emmartins** [comment]
retest this please

File: controller/src/main/java/org/jboss/as/controller/security/ControllerPermission.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+     *
+     * @param name the name of the permission.
+     * @param actions should be null.
+     *
+     * @throws NullPointerException if <code>name</code> is <code>null</code>.
+     * @throws IllegalArgumentException if <code>name</code> is empty.
+     */
+    public ControllerPermission(String name, String actions) {
+        super(name, actions);
+    }
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/security/ControllerPermission.java]: Why do we have this constructor?

File: controller/src/main/java/org/jboss/as/controller/security/ControllerPermission.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+ * </p>
+ *
+ * <p>
+ * The naming convention follows the hierarchical property naming convention.
+ * An asterisk may appear by itself, or if immediately preceded by a "."
+ * may appear at the end of the name, to signify a wildcard match.
+ * </p>
+ *
+ * <p>
+ * The target name is the name of the permission. The following table lists all the possible {@link org.jboss.as.controller.security.ControllerPermission} target names,
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/security/ControllerPermission.java]: These are just the well known ones, not "all possible" as the constructor is not private.

File: controller/src/main/java/org/jboss/as/controller/ModelController.java [hunk 3]
```diff
@@ -40,10 +41,10 @@
 public interface ModelController {
 
     /**
-     * A {@link RuntimePermission} needed to access a {@link ModelController} via {@link org.jboss.msc.service.Service#getValue()} or
-     * to invoke its methods. The name of the necessary {@code RuntimePermission} is "{@code canAccessModelController}."
+     * A {@link org.jboss.as.controller.security.ControllerPermission} needed to access a {@link ModelController} via {@link org.jboss.msc.service.Service#getValue()} or
+     * to invoke its methods. The name of the permission is "{@code canAccessModelController}."
      */
-    RuntimePermission ACCESS_PERMISSION = new RuntimePermission("canAccessModelController");
+    ControllerPermission CAN_ACCESS_MODEL_CONTROLLER = new ControllerPermission("canAccessModelController");
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/ModelController.java]: Please revert the name change. It's not necessary and this is public API. The type change is unavoidable, but it's likely anyone using this constant would only care about the "Permission"  type anyway.

File: controller/src/main/java/org/jboss/as/controller/registry/ImmutableManagementResourceRegistration.java [hunk 4]
```diff
@@ -41,10 +42,10 @@
 public interface ImmutableManagementResourceRegistration {
 
     /**
-     * A {@link RuntimePermission} needed to create a {@link ImmutableManagementResourceRegistration} or invoke one
-     * of its methods. The name of the necessary {@code RuntimePermission} is "{@code canAccessManagementResourceRegistration}."
+     * A {@link org.jboss.as.controller.security.ControllerPermission} needed to create a {@link ImmutableManagementResourceRegistration} or invoke one
+     * of its methods. The name of the permission is "{@code canAccessManagementResourceRegistration}."
      */
-    RuntimePermission ACCESS_PERMISSION = new RuntimePermission("canAccessManagementResourceRegistration");
+    ControllerPermission CAN_ACCESS_MANAGEMENT_RESOURCE_REGISTRATION = new ControllerPermission("canAccessManagementResourceRegistration");
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/registry/ImmutableManagementResourceRegistration.java]: Please revert the name change. It's not necessary and this is public API. The type change is unavoidable, but it's likely anyone using this constant would only care about the "Permission"  type anyway.

File: server/src/main/java/org/jboss/as/server/deployment/service/SecuredServiceRegistry.java [hunk 5]
```diff
@@ -15,7 +16,10 @@
  */
 public class SecuredServiceRegistry implements ServiceRegistry {
 
-    public static final RuntimePermission PERMISSION = new RuntimePermission("org.jboss.as.server.deployment.service.USE_SERVICE_REGISTRY");
+    /**
+     * A {@link org.jboss.as.server.security.ServerPermission} needed to use {@link org.jboss.as.server.deployment.service.SecuredServiceRegistry}, i.e. invoke its methods. The name of the permission is "{@code useServiceRegistry}."
+     */
+    public static final ServerPermission USE_SERVICE_REGISTRY = new ServerPermission("useServiceRegistry");
```
Comment (bstansberry) [inline server/src/main/java/org/jboss/as/server/deployment/service/SecuredServiceRegistry.java]: Please revert the name change. It's not necessary and this is public API. The type change is unavoidable, but it's likely anyone using this constant would only care about the "Permission"  type anyway.

File: server/src/main/java/org/jboss/as/server/security/ServerPermission.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+     *
+     * @param name the name of the permission.
+     * @param actions should be null.
+     *
+     * @throws NullPointerException if <code>name</code> is <code>null</code>.
+     * @throws IllegalArgumentException if <code>name</code> is empty.
+     */
+    public ServerPermission(String name, String actions) {
+        super(name, actions);
+    }
```
Comment (bstansberry) [inline server/src/main/java/org/jboss/as/server/security/ServerPermission.java]: No reason to include an unused constructor.

File: server/src/main/java/org/jboss/as/server/security/ServerPermission.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+ * </p>
+ *
+ * <p>
+ * The naming convention follows the hierarchical property naming convention.
+ * An asterisk may appear by itself, or if immediately preceded by a "."
+ * may appear at the end of the name, to signify a wildcard match.
+ * </p>
+ *
+ * <p>
+ * The target name is the name of the permission. The following table lists all the possible permission target names,
```
Comment (bstansberry) [inline server/src/main/java/org/jboss/as/server/security/ServerPermission.java]: This isn't all possible permission targets, as the constructor is public. It's the well known ones.

File: controller/src/main/java/org/jboss/as/controller/security/ControllerPermission.java [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+ * <table border=1 cellpadding=5 summary="permission target name,
+ *  what the target allows">
+ * <tr>
+ * <th>Permission Target Name</th>
+ * <th>What the Permission Allows</th>
+ * </tr>
+ *
+ * <tr>
+ *   <td>canAccessManagementResourceRegistration</td>
+ *   <td>Creation of {@link org.jboss.as.controller.registry.ManagementResourceRegistration}, or invoke one
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/security/ControllerPermission.java]: also the ImmutableManagementResourceRegistration super-interface

File: controller/src/main/java/org/jboss/as/controller/security/ControllerPermission.java (hunk 2 above)
Comment (dmlloyd) [inline controller/src/main/java/org/jboss/as/controller/security/ControllerPermission.java]: Having non-finite targets precludes efficient permission collections.  That said, this implementation is clearly not designed for that.  But the target (and action) strings should be validated regardless.

File: controller/src/main/java/org/jboss/as/controller/security/ControllerPermission.java (hunk 1 above)
Comment (dmlloyd) [inline controller/src/main/java/org/jboss/as/controller/security/ControllerPermission.java]: You have to have both constructors else the security policy code will be angry.

Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/security/ControllerPermission.java]: Ok; good to know. SecurityManager stuff is such a joy.

Comment (dmlloyd) [inline controller/src/main/java/org/jboss/as/controller/security/ControllerPermission.java]: It truly is.

**Comment by emmartins** [comment]
update changes:
1. public api permission instance names back to original one
2. permission names and actions validated

**Comment by bstansberry** [comment]
I rebased this and opened #466 with the rebase. I'll merge that once the PR run is done.

**Comment by bstansberry** [comment]
#466 is merged.

**Pull request closed without merging**